src/Symfony/Component/Form/Tests/Extension/Core/Type/ChoiceTypeTest.php

```diff
@@ -1247,7 +1279,7 @@ public function testSubmitSingleExpandedWithEmptyChild()
 
         $form->submit('');
 
-        $this->assertNull($form->getData());
+        $this->assertEmpty($form->getData());
```

Comment:
@webmozart you need to read this test in particular, the others are just using an empty string instead of `null` as `viewValue` but this one concerns the data.

Also this PR fixes the failed tests added originally in #17760 (ref https://github.com/symfony/symfony/pull/17760/files#diff-533d45d8bb41e489890ad12df7a1a4b7R163).